Core/automation/lib/python/core/utils.py

```diff
@@ -118,6 +122,61 @@ def validate_uid(uid):
     return uid
 
+def get_item_value(item_or_item_name, return_type=None, default=None, unit=None):
+    """
+    Returns the Item's value if the Item exists and is initialized, otherwise default. 
+    Returned item type will be dependent on the item's type. If return_type is supplied,
+    the Item will be returned as that type if possible, otherwise the default will be 
+    returned. Item value will be converted to the provided unit type if possible; otherwise,
+    the default will be returned. If no default is supplied and an Item value cannot be 
+    returned, None is returned 
+
+    Args:
+        item_or_item_name (Item or str): name of the Item
+        return_type (int, float, str): item type to be returned. If the Item cannot be 
+            converted to return_type, default will be returned 
+        default (int, float, ON, OFF, OPEN, CLOSED, str, DateTime): the default
+            value to be returned if the Item is non-existent/uninitialized or cannot
+            be converted to return_type
+        unit (QuantityType): Convert value of Item to specified units
+            https://www.openhab.org/docs/concepts/units-of-measurement.html
+
+    Returns:
+        int, float, ON, OFF, OPEN, CLOSED, str, DateTime, or None: the state of the Item
+            as a suitable type, or converted to return_type if given; or the default value 
+            if the Item's state cannot be converted or is NULL or UNDEF; otherwise None
+    """
+    
+    type_methods = {int: "intValue", float: "floatValue", str: "toString"}
+    item = validate_item(item_or_item_name) # type: t.Union[GenericItem, None]
+    if item:
+        state = item.getState() # type: State
+        if isinstance(state, QuantityType) and unit is not None:
+            try:
+                state = state.toUnit(unit)
+            except:
+                LOG.warn(u"Item '{}' cannot be converted to '{}' units, returning default value".format(item.name, unit))
+                return default
+        if not isinstance(state, UnDefType):
+            if return_type is None:
+                return state
+            elif return_type in type_methods:
+                if hasattr(state, type_methods[return_type]):
+                    state = getattr(state, type_methods[return_type])()
+            try: # attempt to cast
+                return return_type(state)
+            except:
+                LOG.warn(u"Item '{}' is a '{}', cannot be converted to '{}' type, returning default value".format(item.name, type(state), return_type))
+                pass
+    else:
+        LOG.warn(u"Item does not exist or is uninitialized, returning default value")
+    return default
+
+def get_item_value_or_default(item_or_item_name, default_value, return_type=None, unit=None):
+    """
+    See get_item_or_default, this is a drop-in replacement for old getItemValue
+    """
+    return get_item_value(item_or_item_name, return_type=return_type, default=default_value, unit=unit)
```

Comment:
I disagree with adding this, the new function does this out of the box. Also the docstring here is a lie, this function *cannot* be used as a drop-in replacement because it doesn't return the same result given the same inputs.

I do agree about the order of arguments, thinking about it a bit more it seems like a default would be provided more often that a return type. On that thought, what about `get_item_state` as a function name? Gives a better idea of what it does.

Reply:
Yes, I'm discovering this as I attempt to use it as such.

Reply:
> I was trying to do similar to

The example you show does work the same, in fact only the name was changed in order to make everything snake_case and provide clearer function names.

Reply:
Removed